test/helpers/IntegrationHelper.js

```diff
@@ -754,6 +754,42 @@ function generatePromptHelpMessage(player) {
     return `Current prompt for ${player.name}:\n${formatPrompt(player.currentPrompt(), player.currentActionTargets)}`;
 }
 
+function validatePlayerOptions(playerOptions, playerName, startPhase) {
+    // list of approved property names
+    const listOfApprovedPropertiesForAnyOtherPhase = [
+        'hasInitiative',
+        'resources',
+        'groundArena',
+        'spaceArena',
+        'hand',
+        'discard',
+        'leader',
+        'base',
+        'deck',
+        'resource',
+    ];
+    // list of approved property names for setup phase
+    const listOfApprovedPropertiesForSetupPhase = [
+        'leader',
+        'deck'
+    ];
+
+    // Check for unknown properties
+    if (startPhase !== 'setup') {
+        Object.keys(playerOptions).forEach((prop) => {
```

Comment:
Per the discussion in discord, for anything that's more than a one-liner I would prefer to use `for` loops since it's better for debugging (the vscode debugger skips directly over forEach loops). For one-line statements I'm more okay with forEach but there's still some discussion on it

Reply:
i also made it a bit more clearer and cleaner.